### PR TITLE
now it works with back for sure

### PR DESCRIPTION
**Pull request by tolyanchenko**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,31 +1,60 @@
...
+  const resp = await fetch("http://localhost:8000/allTasks", {
+    method: "GET",
+  });
+  const result = await resp.json();
+  allTasks = result.data;
   render();
 };
 
-pushButton = () => {
+async function serverRender() {
```
Comment (usatovaDasha) [inline test.js:17]: es6 function

File: test.js [hunk 2]
```diff
@@ -1,31 +1,60 @@
...
+  input.addEventListener("change", updateValue);
+  const resp = await fetch("http://localhost:8000/allTasks", {
+    method: "GET",
+  });
+  const result = await resp.json();
+  allTasks = result.data;
+  render();
+}
+
+onClickButton = async () => {
```
Comment (usatovaDasha) [inline test.js:28]: const onClickButton =

File: test.js [hunk 3]
```diff
@@ -1,31 +1,60 @@
...
   allTasks.push({
```
Comment (usatovaDasha) [inline test.js:29]: no need this lines when you use server

File: test.js [hunk 4]
```diff
@@ -1,31 +1,60 @@
...
+  });
+  const result = await response.json();
+  allTasks = result.data;
+  localStorage.setItem("todoList", JSON.stringify(allTasks));
   valueInput = "";
   input.value = "";
+  render();
 };
 
 updateValue = (event) => {
```
Comment (usatovaDasha) [inline test.js:52]: const updateValue =

File: test.js [hunk 5]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
       checkbox.className = "disabled";
 
-      let newInput = document.createElement("input");
+      const newInput = document.createElement("input");
       newInput.type = "text";
       newInput.value = item.text;
       newInput.id = "inputEdit";
-      newInput.addEventListener("change", startEditTask);
-      // newInput.addEventListener("blur", stopEditTask);
+      newInput.onchange = editFunction = () => {
```
Comment (usatovaDasha) [inline test.js:116]: newInput.onchange = () => {
        editServerTask(event);
      }

File: test.js [hunk 6]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
+      };
       container.appendChild(newInput);
 
-      let newButton1 = document.createElement("button");
+      const newButton1 = document.createElement("button");
 
       newButton1.innerText = "Accept";
       newButton1.id = "newButton";
-      newButton1.addEventListener("click", stopEditTask);
+      newButton1.onclick = stopEditFunction = () => {
```
Comment (usatovaDasha) [inline test.js:125]: higher

File: test.js [hunk 7]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
     container.appendChild(deleteDiv);
 
     content.appendChild(container);
-    console.clear;
   });
 };
 
-onChangeCheckBox = (index) => {
-  allTasks[index].isCheck = !allTasks[index].isCheck;
+forDeleteAllButton = async (index) => {
```
Comment (usatovaDasha) [inline test.js:160]: delete all or delete one?
and you don't check answer from server?

File: test.js [hunk 8]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
+  const response = await fetch(
+    `http://localhost:8000/deleteTask?id=${allTasks[index].id}`,
+    {
+      method: "DELETE",
+    }
+  );
+};
+
+pushDeleteAllButton = async () => {
```
Comment (usatovaDasha) [inline test.js:169]: no need async

File: test.js [hunk 9]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
+  allTasks.map((item, index) => {
+    forDeleteAllButton(index);
+  });
+  allTasks = [];
   render();
 };
```
Comment (usatovaDasha) [inline test.js:175]: after } ; not put

File: test.js [hunk 10]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
 
-startEditTask = (event) => {
+deleteServerTask = async (index) => {
```
Comment (usatovaDasha) [inline test.js:177]: and you don't check answer from server?

File: test.js [hunk 11]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
 };
 
-cancelEditTask = (event) => {
-  allTasks[editTask].text = saveTasks;
-  stopEditTask();
+stopEditServerTask = async (index) => {
+  const response = await fetch(`http://localhost:8000/updateTask`, {
+    method: "PATCH",
+    body: JSON.stringify({
+      id: allTasks[index].id,
```
Comment (usatovaDasha) [inline test.js:195]: using destructurization

File: test.js [hunk 12]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
+      text: allTasks[editTask].text,
+    }),
+    headers: {
+      "Content-type": "application/json; charset=UTF-8",
+    },
+  });
+  const result = await response.json();
+  localStorage.setItem("todoList", JSON.stringify(allTasks));
```
Comment (usatovaDasha) [inline test.js:203]: no need change allTasks?

File: test.js [hunk 13]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
 
-pushDeleteAllButton = () => {
-  allTasks = [];
-  localStorage.setItem("todoList", JSON.stringify(allTasks));
-  render();
+onChangeCheckBox = async (index) => {
+  const response = await fetch(`http://localhost:8000/updateTask`, {
+    method: "PATCH",
+    body: JSON.stringify({
+      id: allTasks[index].id,
```
Comment (usatovaDasha) [inline test.js:218]: destructurization

File: test.js [hunk 14]
```diff
@@ -80,75 +109,119 @@ render = () => {
...
+      isCheck: !allTasks[index].isCheck,
+    }),
+    headers: {
+      "Content-type": "application/json; charset=UTF-8",
+    },
+  });
+  const result = await response.json();
```
Comment (usatovaDasha) [inline test.js:225]: no need change allTasks?

File: test.js (hunk 9 above)
Comment (tolyanchenko) [reply test.js:175]: But VS code does it itself, really! It's not like I really wanted to be doing it myself...

File: test.js [hunk 15]
```diff
@@ -14,22 +13,16 @@ window.onload = async () => {
   render();
 };
 
-async function serverRender() {
-  input = document.getElementById("textId");
-  input.addEventListener("change", updateValue);
+serverRender = async () => {
```
Comment (usatovaDasha) [inline test.js:16]: const

**Pull request merged**